src/main/java/org/dependencytrack/tasks/metrics/ComponentMetricsUpdateTask.java

```diff
@@ -96,13 +96,16 @@ static Counters updateMetrics(final UUID uuid) throws Exception {
                         .forEach(aliasesSeen::add);
 
                 counters.vulnerabilities++;
-
-                switch (vulnerability.getSeverity()) {
-                    case CRITICAL -> counters.critical++;
-                    case HIGH -> counters.high++;
-                    case MEDIUM -> counters.medium++;
-                    case LOW, INFO -> counters.low++;
-                    case UNASSIGNED -> counters.unassigned++;
+                try {
+                    switch (vulnerability.getSeverity()) {
+                        case CRITICAL -> counters.critical++;
+                        case HIGH -> counters.high++;
+                        case MEDIUM -> counters.medium++;
+                        case LOW, INFO -> counters.low++;
+                        case UNASSIGNED -> counters.unassigned++;
+                    }
+                } catch (NullPointerException ex) {
+                    LOGGER.debug("Vulnerability severity is null for" + vulnerability.getSource() + "|" + vulnerability.getVulnId() + "->"+vulnerability.getSeverity());
```

Comment:
Can we make this an explicit `null` check, and log a warning instead? After the migration to v4.11, severities should no longer be `null`. If they still are, we need to know about it so we can fix it.

Reply:
Sure, let me fix it.